TypeCobol/Tools/Options-Config/TypeCobolConfiguration.cs

```diff
@@ -44,29 +140,183 @@ public static OptionSet GetCommonTypeCobolOptions(TypeCobolConfiguration typeCob
+                { "e|encoding=", "{ENCODING} of the file(s) to parse. It can be one of \"rdz\"(this is the default), \"zos\", or \"utf8\". "+"If this option is not present, the parser will attempt to guess the {ENCODING} automatically.",
+                    v => typeCobolConfig.RawFormat = v},
                 { "y|intrinsic=", "{PATH} to intrinsic definitions to load.\nThis option can be specified more than once.", v => typeCobolConfig.Copies.Add(v) },
                 { "c|copies=",  "Folder where COBOL copies can be found.\nThis option can be specified more than once.", v => typeCobolConfig.CopyFolders.Add(v) },
                 { "dp|dependencies=", "Path to folder containing programs to load and to use for parsing a generating the input program.", v => typeCobolConfig.Dependencies.Add(v) },
                 { "t|telemetry", "If set to true telemetry will send automatic email in case of bug and it will provide to TypeCobol Team data on your usage.", v => typeCobolConfig.Telemetry = true },
-                { "md|maximumdiagnostics=", "Wait for an int value that will represent the maximum number of diagnostics that TypeCobol have to return.", v =>  typeCobolConfig.MaximumDiagnostics = int.Parse(v.ToString())},
-                { "f|outputFormat=", "Output format (default is Cobol 85). (Cobol85/0, PublicSignature/1)", v => Enum.TryParse(v.ToString(), true, out typeCobolConfig.OutputFormat)},
+                { "md|maximumdiagnostics=", "Wait for an int value that will represent the maximum number of diagnostics that TypeCobol have to return.", v =>  typeCobolConfig.RawMaximumDiagnostics = v},
+                { "f|outputFormat=", "Output format (default is Cobol 85). (Cobol85/0, PublicSignature/1)", v =>typeCobolConfig.RawOutputFormat = v},
                 { "ec|expandingcopy=", "Generate a file with all COPY directives expanded in the source code. This option will be executed if the Preprocessor step is enabled.", v => typeCobolConfig.ExpandingCopyFilePath = v },
                 { "exc|extractusedcopy=", "Generate a file with all COPIES detected by the parser", v => typeCobolConfig.ExtractedCopiesFilePath = v },
-            };
 
+            };
             return commonOptions;
         }
 
+        public static Dictionary<ReturnCode, string> InitializeCobolOptions(TypeCobolConfiguration config, IEnumerable<string> args, OptionSet options)
+        {
+            Dictionary<ReturnCode, string> errorStack = new Dictionary<ReturnCode, string>();
+            List<string> unexpectedStrings = new List<string>();
+            
+            try
+            {
+                unexpectedStrings = options.Parse(args);
+            }
+            catch (OptionException ex)
+            {
+                // Last parameter doesn't have any value
+                errorStack.Add(ReturnCode.FatalError, ex.Message);
+            }
+
+
+            // ExecToStepError
+            switch (config.RawExecToStep.ToLower())
```

Comment:
Why not using this code Enum.TryParse(v.ToString(), true, out typeCobolConfig.ExecToStep) to convert the string to an Enum value ? You should also be able to get the exception